src/middleware.js

```diff
@@ -12,18 +12,16 @@ class KoaJoiValidatorMiddleware {
     this.getSubject = getSubject;
 
     this._validateParameters();
+
+    this.compiledSchema = Joi.compile(schema);
   }
 
   async execute(ctx, next) {
-    this.ctx = ctx;
-    this.next = next;
-    try {
```

Comment:
If we refactor the try block to use .validate instead, only ValidationError will be returned and other errors will not be caught. This makes the `if (error instanceof Joi.ValidationError === false) {` line unnecessary, also removes the ability to handle other errors, like when the error is originating from the `this.getSubject(ctx);` function call.